### DISCUSSION TITLE
Are Cordova WebViews different from system WebViews?

### DISCUSSION THREAD

**Discussion by jmarshall-com** [apache/cordova#356] [Q&A] [answered]
New to mobile development here.  Thanks for Cordova, and for starting discussions on GitHub.  :)

I would love to better understand the relationship between the WebViews used by Cordova, and the WebViews provided by the Android and iOS systems.  In particular, I couldn't get the JS "import" statement to be recognized, as if the Cordova WebViews used an older version of JS.  Is this true?  If so, it becomes a problem when trying to use various JS libraries in Cordova apps.  Or am I getting something wrong?  Thanks!

**Comment by erisu** [comment, marked as answer]
Cordova does not have any custom WebViews. It uses the System WebView provided by Android and iOS.

Per [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import), the `import` statements were introduced in Chrome 80 and Safari 15.

[Since Cordova-Android 9.x release, the defined minimum supported SDK is 22 (Android 5.1).](https://cordova.apache.org/docs/en/11.x/guide/platforms/android/index.html#android-api-level-support)

Starting in [Android 5.0 (SDK 21), the WebView was moved to an APK so it can be updated separately to the platform](https://developer.chrome.com/docs/multidevice/webview/). End-users should be able to upgrade to the latest version of Chromium that supports the `import` statement. But this relies on the end-user to update their browser regularly. If you change the minimum target to something lower than SDK 21, those users will have issues with the `import` statement.

As for iOS users, [Safari 15 was shipped with iOS and iPadOS 15.](https://developer.apple.com/documentation/safari-release-notes/safari-15-release-notes) Since Safair version is tied to the platform's version, anyone running older iOS version will have issues with the import statement.

**Reply by jmarshall-com** [reply]
Excellent, thank you for the detailed explanation!

**Reply by breautek** [reply]
To add onto the answer. SDK 22, or Android 5.1 if you create an AOSP emulator and don't update, the installed webview version is Chrome 33 (if I recall correctly). Which means a Cordova app running on Android 5.1, could potentially be running a webview version as old as Chrome 33, assuming that the user never updated their device from factory settings.
